source/s3_meta_request.c

```diff
@@ -1062,6 +1056,17 @@ static void s_s3_meta_request_request_on_signed(
             metric->time_metrics.sign_end_timestamp_ns - metric->time_metrics.sign_start_timestamp_ns;
     }
 
+    /**
+     * Add "x-amz-content-sha256: STREAMING-UNSIGNED-PAYLOAD-TRAILER" header to support trailing checksum.
+     */
+    if (request->send_data.require_streaming_unsigned_payload_header) {
+        struct aws_http_headers *headers = aws_http_message_get_headers(request->send_data.message);
+        error_code = aws_http_headers_set(
```

Comment:
3 things:
1) bug: if there's an error here, `error_code` will be -1, but it's *supposed* to be the actual error code
2) style: please add an explicit `goto finish` after setting error_code. I know it's not technically necessar because this is currently the last thing that can go wrong, but if some future person inserts more logic after this if-branch, they might miss that this if-branch would need a goto
3) utterly trivial: move this so it happens before `signing_duration_ns` gets set, since that seems like the last thing that should happen